compiler/cpp/src/generate/t_cpp_generator.cc

```diff
@@ -1506,24 +1523,16 @@ void generate_fields(std::ofstream& out,
 /**
  * Generates operator<<
  */
-void t_cpp_generator::generate_struct_ostream_operator(std::ofstream& out, t_struct* tstruct) {
+void t_cpp_generator::generate_struct_print_operator(std::ofstream& out, t_struct* tstruct) {
   out << indent();
-  generate_struct_ostream_operator_decl(out, tstruct);
+  generate_struct_print_operator_decl(out, tstruct);
   out << " {" << endl;
 
   indent_up();
 
   out << indent() << "using apache::thrift::to_string;" << endl;
-
-  // eliminate compiler unused warning
-  const vector<t_field*>& fields = tstruct->get_members();
-  if (fields.empty())
-    out << indent() << "(void) obj;" << endl;
```

Comment:
why this line is removed? I'm pretty sure struct/exception without fields will generate warning "unused function parameter" for operator<</print method